plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/group/MavenGroupHandler.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal.group;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.group.GroupFacet;
+import org.sonatype.nexus.repository.http.HttpResponses;
+import org.sonatype.nexus.repository.http.HttpStatus;
+import org.sonatype.nexus.repository.maven.internal.context.Coordinates;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Handler;
+import org.sonatype.nexus.repository.view.Request;
+import org.sonatype.nexus.repository.view.Response;
+import org.sonatype.nexus.repository.view.ViewFacet;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import static org.sonatype.nexus.repository.http.HttpMethods.GET;
+import static org.sonatype.nexus.repository.maven.internal.ContextHelper.coordinates;
+import static org.sonatype.nexus.repository.maven.internal.ContextHelper.name;
+
+/**
+ * Maven group handler.
+ *
+ * @since 3.0
+ */
+// TODO: this is started from copy+paste of GroupHandler, but that class as-is cannot be extended to suit this use case
```

Comment:
What use case are you talking about?  We don't want to rebuild mirror copies of things if there is an issue with the underlying common bits like this if we can avoid it.